genesyscloud/tfexporter/genesyscloud_resource_exporter.go

```diff
@@ -386,6 +400,87 @@ func (g *GenesysCloudResourceExporter) generateOutputFiles() diag.Diagnostics {
 	return nil
 }
 
+func (g *GenesysCloudResourceExporter) buildAndExportDependsOnResources() diag.Diagnostics {
+
+	filterList, err := g.processAndBuildDependencies()
+	if err != nil {
+		return err
+	}
+	if len(filterList) > 0 {
+		diagErr := g.exportDependentResources(filterList)
+		if diagErr != nil {
+			return diagErr
+		}
+	}
+	return nil
+}
+
+func (g *GenesysCloudResourceExporter) processAndBuildDependencies() (filters []string, diagErr diag.Diagnostics) {
+	filterList := make([]string, 0)
+	proxy := dependentconsumers.GetDependentConsumerProxy(nil)
+
+	retrieveDependentConsumers := func(resourceKeys resourceExporter.ResourceInfo) func(ctx context.Context, clientConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
+		return func(ctx context.Context, clientConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
+			proxy = dependentconsumers.GetDependentConsumerProxy(clientConfig)
+			resources := make(resourceExporter.ResourceIDMetaMap)
+			resources, err := proxy.GetDependentConsumers(ctx, resourceKeys)
+
+			if err != nil {
+				return nil, diag.Errorf("Failed to retrieve Dependent Flows %s: %s", resourceKeys.State.ID, err)
+			}
+			return resources, nil
+		}
+	}
+
+	for _, resourceKeys := range g.resources {
+
+		resources, err := proxy.GetAllWithPooledClient(retrieveDependentConsumers(resourceKeys))
+		if err != nil {
+			return nil, err
+		}
+
+		if len(resources) > 0 {
+			resourcesTobeExported := retrieveExportResources(g.resources, resources)
+			dependsList := make([]string, 0)
+			for _, meta := range resourcesTobeExported {
+				resource := strings.Split(meta.Name, " ")
+				filterList = append(filterList, fmt.Sprintf("%s::%s", resource[0], resource[1]))
+				dependsList = append(dependsList, fmt.Sprintf("%s.%s", resource[0], resource[1]))
+			}
+
+			dependsMap := make(map[string][]string)
+			dependsMap[resourceKeys.State.ID] = dependsList
```

Comment:
is there any chance that `resourceKeys` or `resourceKeys.State` or `resourceKeys.State.ID` could be nil?